### PR TITLE
Hca store

### PR DESCRIPTION
**Pull request by parthshahva**
Heavily borrowed from https://github.com/HumanCellAtlas/hca-matrix-utils. Importing in hca store, util functions, and tests

### CODE REVIEW

File: matrix/common/util.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import pandas
+import hca
+import zarr
+
+
+def get_client(dss_instance=None):
```
Comment (calvinnhieu) [inline matrix/common/util.py]: What do you think of having this as a @property in `HCAStore`? I think it comes down to, do we know if other parts of matrix service are going to need to access the DSSClient outside of `HCAStore`?

File: requirements.txt [hunk 2]
```diff
@@ -6,3 +6,7 @@ itsdangerous==0.24
 MarkupSafe==1.0
 PyYAML==3.13
 requests==2.19.1
+numpy==1.15.2
+pytest==3.8.2
```
Comment (calvinnhieu) [inline requirements.txt]: requirements-dev.txt?

File: tests/unit/__init__.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+test_bundle_spec = {
```
Comment (calvinnhieu) [inline tests/unit/__init__.py]: I think this is fine, but also that storing this in a file in `matrix/tests/res/` would be cleaner. Fine either way

File: tests/unit/common/test_hca_store.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import pytest
+import zarr
+
+from matrix.common.hca_store import HCAStore
+from .. import test_bundle_spec
+
+TEST_DSS_HOST = 'https://dss.integration.data.humancellatlas.org/v1'
+
+
+class TestLambdaHandler(unittest.TestCase):
```
Comment (calvinnhieu) [inline tests/unit/common/test_hca_store.py]: Name

File: tests/unit/common/test_util.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import numpy
+import unittest
+import pytest
+
+from matrix.common.util import get_client, convert_zarr_store_to_pandas_df
+from matrix.common.hca_store import HCAStore
+from .. import test_bundle_spec
+
+
+class TestUtils(unittest.TestCase):
```
Comment (calvinnhieu) [inline tests/unit/common/test_util.py]: nit: `TestUtil` to match file name

File: matrix/common/util.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    client = hca.dss.DSSClient()
+    if not dss_instance:
+        client.host = "https://dss.integration.data.humancellatlas.org/v1"
+    else:
+        client.host = "https://dss.{dss_instance}.data.humancellatlas.org/v1".format(
+            dss_instance=dss_instance)
+    return client
+
+
+def convert_zarr_store_to_pandas_df(zarr_store, concat=False):
```
Comment (calvinnhieu) [inline matrix/common/util.py]: Function signature types, + others

File: tests/unit/common/test_hca_store.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+    def test_hca_store_read(self):
```
Comment (calvinnhieu) [inline tests/unit/common/test_hca_store.py]: Could use more failure case tests, but would require better error handling. I'm not sure we understand this code path well enough to implement all of those. I'm OK with adding to these tests once we have v0 running.

File: matrix/common/hca_store.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+"""Classes for working with expression matrices serialized in the DCP."""
+
+from collections import MutableMapping
+import hashlib
+
+
+class HCAStore(MutableMapping):
```
Comment (calvinnhieu) [inline matrix/common/hca_store.py]: What do you think of `HCAZarrStore` to communicate that it's zarr compatible?

**Comment by parthshahva** [comment]
Thanks for the comments @calvinnhieu. You did the dirty work of digging through the copy and paste job. I was thinking we could clean it up after getting Marcus' initial implementation merged in but now agree with all your comments and will push up before merging.

**Comment by calvinnhieu** [comment]
Sounds good!

File: matrix/common/hca_store.py (hunk 8 above)
Comment (calvinnhieu) [reply matrix/common/hca_store.py]: Prefer `DSSZarrStore`

**Pull request merged**